lib/src/puppetdb-cli.cc

```diff
@@ -31,49 +33,111 @@ version()
 
 string
 read_config(const string& config_path) {
+    const string defaulted_config_path = config_path.empty() ?
+            "~/.puppetlabs/client-tools/puppetdb.conf" : config_path;
     const string expanded_config_path
-    { futil::tilde_expand("~/.puppetlabs/client-tools/puppetdb.conf") };
+    { futil::tilde_expand(defaulted_config_path) };
     return fs::exists(expanded_config_path) ?
             futil::read(expanded_config_path) : "";
 }
 
 PuppetDBConn
-parse_config(const string& config_content) {
+parse_config(const string& config_content,
+             const string& urls,
+             const string& cacert,
+             const string& cert,
+             const string& key) {
     const json::JsonContainer raw_config(config_content);
     const auto puppetdb_conn = raw_config.includes("puppetdb") ?
-            PuppetDBConn(raw_config.get<json::JsonContainer>("puppetdb")):
-            PuppetDBConn();
+            PuppetDBConn(raw_config.get<json::JsonContainer>("puppetdb"),
+                         urls, cacert, cert, key):
+            PuppetDBConn(urls, cacert, cert, key);
     if (puppetdb_conn.getServerUrl().empty()) {
         throw std::runtime_error { "invalid `server_urls` in configuration" };
     }
     return puppetdb_conn;
 }
 
+vector<string> split(string str, char delimiter) {
+    vector<string> internal;
+    stringstream ss(str);
+    string tok;
+
+    while (getline(ss, tok, delimiter)) {
+        internal.push_back(tok);
+    }
+
+    return internal;
+}
+
+server_urls_t
+parse_server_urls_str(const string& urls) {
+    return split(urls, ',');
```

Comment:
How sensitive is this mechanism, to spaces before/after the comma? Like if I quote a string on the cli for example, and do this what happens?